inspire/modules/deposit/fields/thesis_date.py

```diff
@@ -0,0 +1,39 @@
+#
+## This file is part of INSPIRE.
+## Copyright (C) 2015 CERN.
+##
+## INSPIRE is free software; you can redistribute it and/or
+## modify it under the terms of the GNU General Public License as
+## published by the Free Software Foundation; either version 2 of the
+## License, or (at your option) any later version.
+##
+## INSPIRE is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+## 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+#
+
+from invenio.modules.deposit.fields import Date
+import datetime
+
+__all__ = ['ThesisDate']
+
+
+class ThesisDate(Date):
+    def process_formdata(self, valuelist):
+        if valuelist:
+            date_str = ' '.join(valuelist)
+            for date_format in self.format:
+                try:
+                    self.data = datetime.datetime.strptime(date_str, date_format).date()
+                except ValueError:
+                    pass
+                else:
+                    break
+            else:
+                self.data = None
+                raise ValueError(self.gettext('Not a valid date value'))
```

Comment:
`process_formdata()` is where `self.format` is checked, otherwise it would raise an exception for normal date validation and it wouldn't accept more than one format.

`else` is used only when `for` doesn't run normally (like when it breaks), reference [here](https://docs.python.org/3/reference/compound_stmts.html#for) ;)